zulipterminal/config/themes.py

```diff
@@ -531,3 +537,38 @@ def theme_with_monochrome_added(theme: ThemeSpec) -> ThemeSpec:
             new_style = style
         updated_theme.append(new_style)
     return updated_theme
+
+
+NEW_THEMES = {
+    "gruvbox_dark": gruvbox,
+    "zt_dark": zt_dark,
+    "zt_light": zt_light,
+    "zt_blue": zt_blue,
+}
+
+
+def generate_theme(theme_name: str, color_depth: int) -> ThemeSpec:
+    theme_styles = NEW_THEMES[theme_name].STYLES
+
+    urwid_theme = []
+    for style_name, style_conf in theme_styles.items():
+        fg, bg = style_conf
+        fg_codes = fg.value.split()
+        bg_codes = bg.value.split()
```

Comment:
You can unpack fg and bg in the `for` statement. It may be clearer not reusing the names later too.

For the fg/bg_codes it does seem like we could unpack them into names instead to improve how the code below reads, ie. avoid the numerical indexing.

This method works fine and we only use it on startup, but note that we split the colors up repeatedly for each style step, so we could improve this later by processing the colors first.

Reply:
Re 1, :+1: 

Re 2, I'm not exactly sure are the number of elements needed to unpack vary if properties are used or not, but I'll check this out.

Re processing colors first, I don't think this takes much time because when the print statement was used. It almost instantly displayed it while loading but the usual things took time, there may be other things to optimize at startup.